runtime-light/coroutine/event.h

```diff
@@ -70,6 +70,8 @@ inline auto event::awaiter::cancel_awaiter() noexcept -> void {
   }
   if (m_prev != nullptr) {
     m_prev->m_next = m_next;
+  } else {
```

Comment:
Let's a description here: `we are the head of the awaiters list, so we need to update the head`